### ISSUE TITLE
issue after os upgrade - truthy.py", line 88, - AttributeError: module 'yaml' has no attribute 'tokens'

### ISSUE THREAD

**Issue by slieberth** [adrienverge/yamllint#142]
Hi all,

thanks for providing yamllint, great tool!

I have integrated yamllint into a webtool, which verifies yaml runbooks. I have a working code, which I am using on multiple hosts (MAC, Ubuntu) ... 

Now ... after upgrade to ubuntu 18.4 on one server I get now an error ... after elaborating for two hours by myself, I dare to ask for support here in this forum:

error message:
  File "/teHttpServer.py", line 1013, in checkYamlFile
    yamlErrors = list(gen)
  File "/usr/local/lib/python3.6/dist-packages/yamllint/linter.py", line 196, in _run
    for problem in get_cosmetic_problems(buffer, conf, filepath):
  File "/usr/local/lib/python3.6/dist-packages/yamllint/linter.py", line 140, in get_cosmetic_problems
    context[rule.ID]):
  File "/usr/local/lib/python3.6/dist-packages/yamllint/rules/truthy.py", line 88, in check
    if isinstance(token, yaml.tokens.ScalarToken):
AttributeError: module 'yaml' has no attribute 'tokens'

code is pretty following the example (it has worked ... ):
 
                    conf = YamlLintConfig('extends: default')
                    f = open(preProcessorResultFilename)
                    gen = linter.run(f, conf)    <--- this line is triggering the error

I think this is a ubuntu18.4/python3.6 module compatibility issue, do you have an idea, what the problem could be?

thanks Stefan

**Comment by adrienverge** [comment]
Hi Stefan,

Thanks for this report.

I'm curious about your webtool, what is it?
Can you point me to the code so I can have a look and debug?

**Comment by slieberth** [comment]
Hi Adrien,
sorry for my misinterpretation, of course you can take a look at the code an debug, please ignore the comment on the private code, that was just for development testing, the webtool code you are asking for ist at: 
https://github.com/slieberth/aioRunbook
regards Stefan

**Comment by slieberth** [comment]
Hi Adrien,
I thought about my requirements, and actually a canonical syntax checking is not required in my case. instead I am only interested in cases, if pyyaml cannot load the file. I realized that simple exception catching is sufficient in my case. 
I will change my code (removing yamllint) with following function:

```python
import yaml

runbookString = """config:
  description : |
    Test Yaml Validation
  expected : |
    PDF output is created.
  preparation : |
    basic Test
  workingDir: ./results_TEST
  steps:
    - record:
        name: login test
          device: 127.0.0.1
        vendor: ubuntu
        method: ssh
        user: slieberth
        password: secret
        commands:
          - ls TA
  pdfOutput:
    template: "./template_v3.tex"
    author: Stefan Lieberth"""


for i,runbookLine in enumerate(runbookString.split("\n")):
    print(f"{i+1}::{runbookLine}")

preProcessorResultFilename = "./preProcessorResultFilename.yml"
fh = open (preProcessorResultFilename, 'w' )
fh.write (runbookString)
fh.close ()

with open(preProcessorResultFilename, 'r') as stream:
    try:
        print(yaml.load(stream))
    except Exception as e: 
        print("cannot load yaml")
        print(e)
```

producing following output:
```
stefans-Air-3:tests slieberth$ python3 yamlSyntaxCheckTest.py
1::config:
2::  description : |
3::    Test Yaml Validation
4::  expected : |
5::    PDF output is created.
6::  preparation : |
7::    basic Test
8::  workingDir: ./results_TEST
9::  steps:
10::    - record:
11::        name: login test
12::          device: 127.0.0.1
13::        vendor: ubuntu
14::        method: ssh
15::        user: slieberth
16::        password: secret
17::        commands:
18::          - ls TA
19::  pdfOutput:
20::    template: "./template_v3.tex"
21::    author: Stefan Lieberth
cannot load yaml
mapping values are not allowed here
  in "./preProcessorResultFilename.yml", line 12, column 17
```
I will close the issue, no further follow-up required actually
Stefan
